runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -64,7 +68,11 @@ public AbstractQueuedStreamView(final CorfuRuntime runtime,
                                     final UUID streamId,
                                     StreamOptions options) {
         super(runtime, streamId, QueuedStreamContext::new);
-        this.options = options;
+        this.streamOptions = options;
+        this.readOptions = ReadOptions.builder()
+                .clientCachable(streamOptions.isCacheEntries())
+                .ignoreTrim(streamOptions.isIgnoreTrimmed())
```

Comment:
StreamOptions here is a subset of ReadOptions. But we keep both of them as the class member. I think we only need to keep one. I mean StreamOptions can be passed in the constructor, but later after ReadOption is constructed, we just need to keep ReadOptions as a member. What do you think?

Reply:
Are you suggesting making the address space aware of the stream options ?

Reply:
I'm suggesting for this class, it might be unnecessary to have both StreamOptions and ReadOptions as it's member field, we only need ReadOptions because it is superset of StreamOptions.

Reply:
Yea, thats cleaner.

Reply:
Is this done?

Reply:
done.